src/Spec2-Core/SpAbstractTreeSingleSelectionMode.class.st

```diff
@@ -20,7 +20,7 @@ SpAbstractTreeSingleSelectionMode class >> on: aPresenter [
 
 { #category : #selection }
 SpAbstractTreeSingleSelectionMode >> clearSelection [
-
+	selection ifEmpty: [ ^ self ].
```

Comment:
it's a @jecisc commit

Reply:
It is indeed to not launch multiple time events when re-assigning the selection. 

Maybe it would be better to put the behaviour in the slot but I wanted to get soon in a state where this PR can be integrated. ("Better is the enemy of good")

I'll add an issue to the Spec repo.

Reply:
https://github.com/pharo-spec/Spec/issues/676